core/src/main/kotlin/Formats/KotlinWebsiteHtmlFormatService.kt

```diff
@@ -179,6 +179,33 @@ open class KotlinWebsiteHtmlOutputBuilder(
         }
     }
 
+    override fun appendSourceLinks(platforms: PlatformsData, appendSourceLink: DocumentationNode.(platform: String?) -> Unit) {
+        // For each platform find a source url.
+        // If several platforms use the same source url, use the source url only for the first platform.
+        // If only one platform has a source url, drop platform information.
+        val usedSourceUrls = mutableSetOf<String>()
+        val platformToSourceUrl = mutableMapOf<String, DocumentationNode>()
+
+        val nonJRE6 = platforms.filterNot { it.key.isJREVersion() && it.key.endsWith("6") }
+        nonJRE6.forEach { (platform, nodes) ->
+            nodes.firstOrNull { node ->
+                node.details(NodeKind.SourceUrl).firstOrNull()?.takeUnless { usedSourceUrls.contains(it.name) }?.also {
+                    usedSourceUrls.add(it.name)
+                } != null
```

Comment:
Minor: perhaps can be simplified to
```suggestion
                node.details(NodeKind.SourceUrl).firstOrNull()?.takeIf { usedSourceUrls.add(it.name) } != null
```

Reply:
done